### PR TITLE
feat(jiawen195): pass in projectID to its google meeting when the project is created

### PR DESCRIPTION
**Pull request by jiawen195**

### CODE REVIEW

File: api/google-manager/services/google-manager.js [hunk 1]
```diff
@@ -48,15 +48,14 @@ class GoogleManager {
...
-
       const group = await admin.groups.insert({
         requestBody: {
           email: this.formatEmail(name),
           name,
           description,
         }
       });
-
+      console.log("inserted the group");
```
Comment (alearm246) [inline api/google-manager/services/google-manager.js:58]: Usually, we want to remove any unnecessary console logs before making prs

File: api/project/models/project.js [hunk 2]
```diff
@@ -5,4 +5,32 @@
...
 
-module.exports = {};
+ module.exports = {
+   /**
+    * Triggered before user creation.
+    */
+   lifecycles: {
+     async afterCreate(result, data) {
+
+       const { model } = data;
```
Comment (alearm246) [inline api/project/models/project.js:15]: it looks like you aren's using model anywhere, so you should remove it

File: extensions/content-manager/controllers/collection-types.js [hunk 3]
```diff
@@ -58,32 +58,10 @@ module.exports = {
 
   async create(ctx) {
     const { userAbility, user } = ctx.state;
-    const { model } = ctx.params;
+    const { model, id } = ctx.params;
```
Comment (alearm246) [inline extensions/content-manager/controllers/collection-types.js:61]: it doesn't look like you're using id, so you should remove it

**Review by alearm246 — CHANGES_REQUESTED** [review]

File: api/project/models/project.js [hunk 4]
```diff
@@ -5,4 +5,30 @@
  * to customize this model
  */
 
-module.exports = {};
+module.exports = {
+  /**
+  * Triggered before user creation.
+  */
+  lifecycles: {
```
Comment (chungthuang) [inline api/project/models/project.js:12]: What is lifecycles?

Comment (jiawen195) [reply api/project/models/project.js:12]: It says "The lifecycle hooks are functions that get triggered when the Strapi queries are called. They will get triggered automatically when you manage your content in the Admin Panel or when you develop custom code using queries.", it is triggered after the project is created, so we can create the google meeting and pass in ProjectID.

File: api/project/models/project.js [hunk 5]
```diff
@@ -5,4 +5,30 @@
...
+    async afterCreate(result, data) {
+
+      const { title, description, team } = data;
+
+      const group = await strapi.services['google-manager'].createGroup(description, title);
+
+      await strapi.services.project.giveTeamGroup(team, group);
+
+      //Lets team@devlaunchers.com be owner of the google group to fix google meets auto admit problem
```
[the remark this answers is not in the document]
Comment (alearm246) [reply api/project/models/project.js:21]: oh we are that's just an old comment that hasn't been updated lol

[the remark this answers is not in the document]
Comment (jiawen195) [reply api/project/models/project.js:21]: Sure

File: api/project/models/project.js [hunk 6]
```diff
@@ -5,4 +5,30 @@
...
+
+      await strapi.services.project.giveTeamGroup(team, group);
+
+      //Lets strapi-svc@devlaunchers.com be owner of the google group to fix google meets auto admit problem
+      await strapi.services['google-manager'].joinGroup(group.id, process.env.DEVLAUNCHERS_GOOGLE_DIRECTORY_JWT_SUBJECT, 'OWNER');
+
+
+      const calendar = await strapi.services['google-manager'].createCalendar(title);
+
+      result.calendarId = calendar.id;
```
Comment (chungthuang) [inline api/project/models/project.js:27]: result is a parameter of the function. Will the caller of `afterCreate` see the added field?

**Review by alearm246 — APPROVED** [review]

**Pull request merged**